src/router/storeSync/LayerParamConfig.class.js

```diff
@@ -110,16 +109,74 @@ function dispatchLayersFromUrlIntoStore(store, urlParamValue) {
                 )
             }
             log.debug(`  Add layer ${parsedLayer.id} to active layers`, layerObject)
+            if (layerObject.customAttributes?.features) {
+                layerObject.customAttributes.features
+                    .toString()
+                    .split(':')
+                    .forEach((featureId) => {
+                        featuresRequests.push(
+                            getFeature(
+                                store.getters.getLayerConfigById(parsedLayer.id),
+                                featureId,
+                                store.state.position.projection,
+                                store.state.i18n.lang
+                            )
+                        )
+                    })
+            }
         }
         return layerObject
     })
+
     promisesForAllDispatch.push(
         store.dispatch('setLayers', { layers: layers, dispatcher: STORE_DISPATCHER_ROUTER_PLUGIN })
     )
+    if (featuresRequests.length > 0) {
+        getAndDispatchFeatures(featuresRequests, store) // all features after promise
```

Comment:
I think you should return all dispatch promises you will trigger inside this and add them to `promisesForAllDispatch`, so that the next line `Promise.all` wait on them too. Otherwise we might create some "funny" race condition here

I see that you made this function asyc, so in essence it already returns a Promise (which you could add to the list). The only issue I see with that is that you do not `await` dispatch calls, so either `await` them, or bundle them together and return them here